libraries/standard/mqtt/utest/mqtt_lightweight_utest.c

```diff
@@ -1354,14 +1354,21 @@ void test_MQTT_DeserializeAck_pingresp( void )
     uint8_t buffer[ 10 ] = { 0 };
 
     /* Bad remaining length. */
+    ( void ) memset( ( void * ) &mqttPacketInfo, 0x00, sizeof( mqttPacketInfo ) );
     mqttPacketInfo.type = MQTT_PACKET_TYPE_PINGRESP;
-    mqttPacketInfo.pRemainingData = buffer;
     mqttPacketInfo.remainingLength = MQTT_PACKET_PINGRESP_REMAINING_LENGTH + 1;
     status = MQTT_DeserializeAck( &mqttPacketInfo, &packetIdentifier, &sessionPresent );
     TEST_ASSERT_EQUAL_INT( MQTTBadResponse, status );
 
-    /* Process a valid PINGRESP. */
+    /* Process a valid PINGRESP with remaining data buffer NULL. */
+    mqttPacketInfo.remainingLength = MQTT_PACKET_PINGRESP_REMAINING_LENGTH;
```

Comment:
Is it possible that the buffer would be NULL but the remaining length be > 0? If not, should there be an internal assert to check that the `remainingLength` value is 0 for a `NULL` buffer?

Reply:
It is already checked here. https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/development/libraries/standard/mqtt/src/mqtt_lightweight.c#L1158. MQTT_PACKET_PINGRESP_REMAINING_LENGTH is 0